### ISSUE TITLE
WCAG 2.1 F63 and WCAG 2.0 F63 have different test procedures

### ISSUE THREAD

**Issue by dd8** [w3c/wcag#1047]
https://www.w3.org/TR/WCAG20-TECHS/F63.html says

> Check whether the context is contained in the same sentence, paragraph, list item, table cell, associated table headers, **or preceding heading**.

https://www.w3.org/WAI/WCAG21/Techniques/failures/F63 says

> Check whether the context is contained in the same sentence, paragraph, list item, table cell, or associated table headers.

There is a related technique in WCAG 2.1 marked as advisory, but marked as sufficient in 2.0:

https://www.w3.org/TR/WCAG20-TECHS/H80.html
https://www.w3.org/WAI/WCAG21/Techniques/html/H80

**Comment by mraccess77** [comment]
This change/removal of headings occurred several years ago so it sounds like an older version got resurrected in 2.0.

**Comment by awkawk** [comment]
In WCAG 2.0 H80 is marked Advisory: https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0&showtechniques=244

https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-refs.html

My historical reconstruction of the events, from 6 years ago: https://lists.w3.org/Archives/Public/w3c-wai-gl/2014JulSep/0122.html

**Comment by JAWS-test** [comment]
The technique of determining the link context from the previous heading has been removed with reference to the lack of support in JAWS. However, this was incorrect because JAWS has several methods of transmitting the link context based on the heading that are probably better known than the methods for other contexts, such as
* INS+T (reads the page title first and then the heading)
* INS+F6 (focuses the previous heading in the list of headings).

It is questionable whether the technology exists for other screen readers. However, it is also questionable whether all screen readers can determine the link context for the other methods.

Furthermore, I think the criterion is problematic anyway, because a meaningful link label is also important for visually impaired and cognitively impaired people. These users do not have the screen reader methods.

**Comment by awkawk** [comment]
@JAWS-test it wasn't the lack of support in JAWS, it was the lack of support in anything other than JAWS.

**Comment by JAWS-test** [comment]
@awkawk I just followed your links and found that it was the lack of support from JAWS: https://lists.w3.org/Archives/Public/public-comments-wcag20/2012Apr/0031.html.
Also, the techniques always referred only to JAWS: https://www.w3.org/TR/2012/NOTE-WCAG20-TECHS-20120103/H80.

Has anyone ever checked if the other screen readers support the other methods?

But what I find much more problematic with SC 2.4.4 is that 
* blind users have to find out what context the information contains (which they cannot know in advance)
* Blind users may not be able to determine context in certain circumstances, such as when they use the Link Overview (INS+F7 in JAWS)

**Comment by awkawk** [comment]
hmm... ok, I'll take a look

**Comment by mraccess77** [comment]
In my opinion, the case that screen readers only show the accessible name for a link in the list of links dialog is a choice of the screen reader manufacturer.  All of the other information is programmatically available in a way that can be accessed by the screen reader and could be displayed in the dialog.   To my knowledge they don't show aria-describedby either -- that's their omission and I'd encourage folks to log feature requests with screen reader vendors to implement that.

If any user pulled links out of a page without any context -- all users would have the same challenge.  Understanding what is on a page by pulling up a list of links out of context is not very effective.  The list of links dialog is useful when you know a certain link name and it's an efficient way to navigate to and activate links that you commonly use -- but those are two different use cases.  Unfortunately folks rely on that list of links not just for navigation but also to explore pages -- likely because back in 1998 that along with linearizing a page were the the primary methods of exploration.  However, as screen reader capabilities for navigation have increased there is still a desire to hold on to older methods people used for exploration rather than also using newer approaches that provide more contextual details that weren't there before browser navigation.

When in browser mode desktop screen reader users can use commands to read the context without moving the virtual focus by using commands like say sentence, say paragraph, or say line among others.

**Comment by JAWS-test** [comment]
> In my opinion, the case that screen readers only show the accessible name for a link in the list of links dialog is a choice of the screen reader manufacturer. All of the other information is programmatically available in a way that can be accessed by the screen reader and could be displayed in the dialog. ...  I'd encourage folks to log feature requests with screen reader vendors to implement that.

How should a screen reader know what the context is (sentence, paragraph, list item, table cell etc.) when determining the purpose of the link?

**Comment by dd8** [comment]
Setting aside the discussion with the techniques, the root cause of the problem is having two different sets of web pages for the same failures and techniques in WCAG 2.0 and 2.1. 

That's going to cause problems (like different test procedures) for users of the techniques who arrive via Google (e.g. by searching for "WCAG F63") and may land on the 2.0 or 2.1 version.

Having to manually backport changes to the WCAG 2.0 docs when a 2.1 doc is fixed will lead to recurring issues, so one of the these options might help:

1) automatically diff the published 2.0 and 2.1 techniques when building the published pages 
2) automatically copy shared 2.0/2.1 techniques to the 2.0 tree when building the published pages

**Comment by mraccess77** [comment]
This issue is the same for all users- usually people might read what comes before the link on the same line. Then they may read what comes after the link on the same line. Then they might expand that to the sentence or paragraph if needed.

**Comment by JAWS-test** [comment]
> This issue is the same for all users- usually people might read what comes before the link on the same line. Then they may read what comes after the link on the same line. Then they might expand that to the sentence or paragraph if needed.

I don't think so. Because when I work with a screen magnifier, for example, I have a greater effort to determine the context. This also applies to reading with a screen reader. It takes much longer to determine the context with different screen reader methods than to get a visual impression of the context.

**Comment by mraccess77** [comment]
In my opinion turning large pieces of content into links with underlines would actually make things more difficult for users with low vision.  Asking AT vendors to announce or highlight content associated via aria-describedby seems reasonable given that information is available programmatically.

**Comment by alastc** [comment]
> the root cause of the problem is having two different sets of web pages for the same failures and techniques in WCAG 2.0 and 2.1.

Worth noting that we aim to have one set of associated docs for all 2.x versions. We are not there yet, but anything at https://www.w3.org/TR/WCAG20-TECHS/ will be forwarded (or possibly marked as superceded).

**Comment by goodwitch** [comment]
I look forward to this being resolved.  The difference in F63 test procedures in WCAG 2.0 and WCAG 2.1 has caused some confusion for one of my employees...and it took me a while to notice track down the root cause (and confirm which version of F63 is the right one to use).  

I also wanted to add...how very grateful I am that we are now in github because it is now possible for anyone in the world to see exactly when changes were made.  So, thank y'alll.

**Comment by dd8** [comment]
On a related note the UA notes for WCAG 2.0 H81 say it's not actually supported in any AT - although the UA notes have been removed from the WCAG 2.1 version of the H81:
https://www.w3.org/TR/WCAG20-TECHS/H81.html#ua2.46.1

If it's still not supported by AT that seems very similar to relying on previous heading for context.

**Comment by kfranqueiro** [comment]
> Worth noting that we aim to have one set of associated docs for all 2.x versions. We are not there yet, but anything at https://www.w3.org/TR/WCAG20-TECHS/ will be forwarded (or possibly marked as superceded).

Extremely belated update: the 2.0 docs are now marked as superseded. https://www.w3.org/TR/WCAG20-TECHS/F63.html includes a message linking to the actively-maintained 2.2 version of the page.

Since it sounds like that was the primary issue, I'm going to close this.
